packages/realm/src/TypeHelpers.ts

```diff
@@ -16,426 +16,60 @@
 //
 ////////////////////////////////////////////////////////////////////////////
 
-import {
-  BSON,
-  ClassHelpers,
-  Collection,
-  Counter,
-  Dictionary,
-  INTERNAL,
-  List,
-  ObjCreator,
-  PresentationPropertyTypeName,
-  REALM,
-  Realm,
-  RealmObject,
-  RealmSet,
-  TypeAssertionError,
-  UpdateMode,
-  assert,
-  binding,
-  boxToBindingGeospatial,
-  circleToBindingGeospatial,
-  createDictionaryAccessor,
-  createListAccessor,
-  isGeoBox,
-  isGeoCircle,
-  isGeoPolygon,
-  polygonToBindingGeospatial,
-  safeGlobalThis,
-} from "./internal";
-
-const TYPED_ARRAY_CONSTRUCTORS = new Set(
-  [
-    DataView,
-    Int8Array,
-    Uint8Array,
-    Uint8ClampedArray,
-    Int16Array,
-    Uint16Array,
-    Int32Array,
-    Uint32Array,
-    Float32Array,
-    Float64Array,
-    // These will not be present on old versions of JSC without BigInt support.
-    safeGlobalThis.BigInt64Array,
-    safeGlobalThis.BigUint64Array,
-  ].filter((ctor) => ctor !== undefined),
-);
-
-export function toArrayBuffer(value: unknown, stringToBase64 = true) {
-  if (typeof value === "string" && stringToBase64) {
-    return binding.Helpers.base64Decode(value);
-  }
-  for (const TypedArray of TYPED_ARRAY_CONSTRUCTORS) {
-    if (value instanceof TypedArray) {
-      return value.buffer.slice(value.byteOffset, value.byteOffset + value.byteLength);
-    }
-  }
-  assert.instanceOf(value, ArrayBuffer);
-  return value;
+import { assert, binding } from "./internal";
+
+import { createIntTypeHelpers } from "./type-helpers/Int";
+import { createBoolTypeHelpers } from "./type-helpers/Bool";
+import { createStringTypeHelpers } from "./type-helpers/String";
+import { createDataTypeHelpers } from "./type-helpers/Data";
+import { createDateTypeHelpers } from "./type-helpers/Date";
+import { createMixedTypeHelpers } from "./type-helpers/Mixed";
+import { createObjectIdTypeHelpers } from "./type-helpers/ObjectId";
+import { createFloatTypeHelpers } from "./type-helpers/Float";
+import { createDoubleTypeHelpers } from "./type-helpers/Double";
+import { createObjectTypeHelpers } from "./type-helpers/Object";
+import { createLinkingObjectsTypeHelpers } from "./type-helpers/LinkingObjects";
+import { createDecimalTypeHelpers } from "./type-helpers/Decimal";
+import { createUuidTypeHelpers } from "./type-helpers/Uuid";
+import { createArrayTypeHelpers } from "./type-helpers/Array";
+
+import type { TypeHelpers, TypeOptions } from "./type-helpers/types";
```

Comment:
In general, I want to use the `import type` more to avoid circular dependencies at runtime (as these imports are stripped from JS files when running `tsc`). I still have an outstanding task of fixing these circular imports which got leaked into the developers bundler, after we migrated away from Rollup.

Reply:
As you mention, I thought TS handled that implicitly when the import itself was a `type` as in this case. But are you saying that you prefer to still be explicit to be certain?